### PR TITLE
Orthographic projection cli option

### PR DESCRIPTION
**Pull request by kidharb**
Includes

- The command line option
- The test case for the command line option
- Document correction in OPTIONS.md

### CODE REVIEW

File: application/testing/CMakeLists.txt [hunk 1]
```diff
@@ -696,6 +696,7 @@ f3d_test(NAME TestInteractionInvertZoom DATA suzanne.ply ARGS --invert-zoom INTE
 f3d_test(NAME TestInteractionCameraHotkeys DATA cow.vtp INTERACTION DEFAULT_LIGHTS)
 f3d_test(NAME TestInteractionZoomToMouse DATA cow.vtp INTERACTION DEFAULT_LIGHTS)
 f3d_test(NAME TestInteractionOrthographicProjection DATA cow.vtp INTERACTION DEFAULT_LIGHTS) #5;5
+f3d_test(NAME TestInteractionOrthographicProjectionCommand ARGS --camera-orthographic DATA cow.vtp INTERACTION DEFAULT_LIGHTS) #Command line option;
```
Comment (mwestphal) [inline application/testing/CMakeLists.txt:699]: nope, its not an interaction test, just a standard test

File: doc/libf3d/OPTIONS.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ scene.animation.speed-factor|double<br>1<br>render|Set the animation speed facto
 scene.animation.frame-rate|double<br>60<br>render|Set the animation frame rate used to play the animation interactively.|\-\-animation-frame-rate
 scene.camera.index|int<br>-1<br>load|Select the scene camera to use when available in the file.<br>Any negative value means automatic camera.<br>The default scene always uses automatic camera.|\-\-camera-index
 scene.up-direction|string<br>+Y<br>load|Define the Up direction|\-\-up
-camera.orthographic|bool<br>false<br>load|Toggles between orthographic projection and parallel mode.|\-\-orthographic
+scene.camera.orthographic|bool<br>false<br>load|Toggles between orthographic projection and parallel mode.|\-\-orthographic
```
Comment (mwestphal) [inline doc/libf3d/OPTIONS.md:26]: change that in options.cxx as well, and everywhere needed.

File: application/testing/CMakeLists.txt (hunk 1 above)
Comment (kidharb) [reply application/testing/CMakeLists.txt:699]: Offcourse!

File: application/testing/CMakeLists.txt [hunk 3]
```diff
@@ -885,6 +885,9 @@ f3d_test(NAME TestBadRefNoOutput DATA cow.vtp ARGS --ref=${F3D_SOURCE_DIR}/testi
 # Test failure with a bad interaction play file, please do not create a dummy.log
 f3d_test(NAME TestPlayNoFile DATA cow.vtp ARGS --interaction-test-play=${CMAKE_BINARY_DIR}/Testing/Temporary/dummy.log WILL_FAIL)
 
+# Test Orthographic Projection command line option
+f3d_test(NAME TestOrthographicProjectionCommand ARGS --camera-orthographic DATA cow.vtp INTERACTION DEFAULT_LIGHTS) #Command line option;
```
Comment (mwestphal) [inline application/testing/CMakeLists.txt:889]: `f3d_test(NAME TestOrthographicProjection ARGS --camera-orthographic DATA cow.vtp DEFAULT_LIGHTS)`

Put with other classic camera tests, line 156

File: testing/recordings/TestOrthographicProjectionCommand.log [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# StreamVersion 1.2
+ExposeEvent 0 298 0 0 0 0 0
+RenderEvent 0 298 0 0 0 0 0
```
Comment (mwestphal) [inline testing/recordings/TestOrthographicProjectionCommand.log:3]: still not an interaction test, not needed.

Comment (kidharb) [reply testing/recordings/TestOrthographicProjectionCommand.log:3]: I got an error to say that this file was missing when running the CI.
How do I tell the system that it is not an interaction test?

Comment (kidharb) [reply testing/recordings/TestOrthographicProjectionCommand.log]: In the latest CI

```
83/318 Test  #81: f3d::TestOrthographicProjectionCommand .............................***Failed    0.23 sec
Interaction record file to play does not exist /__w/f3d/f3d/source/testing/recordings/TestOrthographicProjectionCommand.log
```
https://github.com/f3d-app/f3d/actions/runs/8314790526/job/22752385391

Comment (kidharb) [reply testing/recordings/TestOrthographicProjectionCommand.log]: Nevermind, I saw the `INTERACTION` in the test case

File: application/F3DOptionsParser.cxx [hunk 5]
```diff
@@ -338,6 +338,7 @@ void ConfigurationOptions::GetOptions(F3DAppOptions& appOptions, f3d::options& o
     this->DeclareOption(grp1, "animation-time", "", "Set animation time to load", options.getAsDoubleRef("scene.animation.time"), HasDefault::YES, MayHaveConfig::YES, "<time>");
     this->DeclareOption(grp1, "animation-frame-rate", "", "Set animation frame rate when playing animation interactively", options.getAsDoubleRef("scene.animation.frame-rate"), HasDefault::YES, MayHaveConfig::YES, "<frame rate>");
     this->DeclareOption(grp1, "font-file", "", "Path to a FreeType compatible font file", options.getAsStringRef("ui.font-file"), LocalHasDefaultNo, MayHaveConfig::NO, "<file_path>");
+    this->DeclareOption(grp1, "camera-orthographic", "", "Render in Orthographic Projection mode", options.getAsBoolRef("scene.camera.orthographic"), LocalHasDefaultNo, MayHaveConfig::YES);
```
Comment (mwestphal) [inline application/F3DOptionsParser.cxx:341]: should go with Camera options and should be labelled something like "Use an orthographic camera"

File: application/testing/CMakeLists.txt [hunk 6]
```diff
@@ -154,6 +154,7 @@ f3d_test(NAME TestColormap DATA IM-0001-1983.dcm ARGS --scalars --roughness=1 --
 f3d_test(NAME TestCameraConfiguration DATA suzanne.obj ARGS --camera-position=0,0,-10 -x --camera-view-up=1,0,0 --camera-focal-point=1,0,0 --camera-view-angle=20 --camera-azimuth-angle=40 --camera-elevation-angle=-80 --camera-direction=12,34,56 --camera-zoom-factor=78)
 f3d_test(NAME TestCameraDirection DATA suzanne.obj ARGS --camera-direction=-1,-1,1 DEFAULT_LIGHTS)
 f3d_test(NAME TestCameraClipping DATA checkerboard_colorful.obj CONFIG ${F3D_SOURCE_DIR}/testing/configs/checkerboard_colorful.json RESOLUTION 800,600)
+f3d_test(NAME TestOrthographicProjectionCommand ARGS --camera-orthographic DATA cow.vtp DEFAULT_LIGHTS) # Test Orthographic Projection command line option
```
Comment (mwestphal) [inline application/testing/CMakeLists.txt:157]: remove the comment, not needed :)

File: doc/libf3d/OPTIONS.md [hunk 7]
```diff
@@ -23,7 +23,7 @@ scene.animation.speed-factor|double<br>1<br>render|Set the animation speed facto
 scene.animation.frame-rate|double<br>60<br>render|Set the animation frame rate used to play the animation interactively.|\-\-animation-frame-rate
 scene.camera.index|int<br>-1<br>load|Select the scene camera to use when available in the file.<br>Any negative value means automatic camera.<br>The default scene always uses automatic camera.|\-\-camera-index
 scene.up-direction|string<br>+Y<br>load|Define the Up direction|\-\-up
-camera.orthographic|bool<br>false<br>load|Toggles between orthographic projection and parallel mode.|\-\-orthographic
+scene.camera.orthographic|bool<br>false<br>load|Toggles between orthographic projection and parallel mode.|\-\-camera\-orthographic
```
Comment (mwestphal) [inline doc/libf3d/OPTIONS.md:26]: please add the doc in `doc/f3d/OPTIONS.md`

File: application/F3DOptionsParser.cxx [hunk 8]
```diff
@@ -392,6 +392,7 @@ void ConfigurationOptions::GetOptions(F3DAppOptions& appOptions, f3d::options& o
     this->DeclareOption(grpCamera, "camera-zoom-factor", "", "Camera zoom factor (non-zero)", appOptions.CameraZoomFactor, HasDefault::YES, MayHaveConfig::YES, "<factor>");
     this->DeclareOption(grpCamera, "camera-azimuth-angle", "", "Camera azimuth angle (in degrees), performed after other camera options", appOptions.CameraAzimuthAngle, HasDefault::YES, MayHaveConfig::YES, "<angle>");
     this->DeclareOption(grpCamera, "camera-elevation-angle", "", "Camera elevation angle (in degrees), performed after other camera options", appOptions.CameraElevationAngle, HasDefault::YES, MayHaveConfig::YES, "<angle>");
+    this->DeclareOption(grpCamera, "camera-orthographic", "", "Use and orthographic camera", options.getAsBoolRef("scene.camera.orthographic"), LocalHasDefaultNo, MayHaveConfig::YES);
```
Comment (mwestphal) [inline application/F3DOptionsParser.cxx:395]: and -> an

Comment (mwestphal) [reply application/F3DOptionsParser.cxx:395]: LocalHasDefaultNo -> HasDefault::YES

File: application/testing/CMakeLists.txt [hunk 9]
```diff
@@ -154,6 +154,7 @@ f3d_test(NAME TestColormap DATA IM-0001-1983.dcm ARGS --scalars --roughness=1 --
 f3d_test(NAME TestCameraConfiguration DATA suzanne.obj ARGS --camera-position=0,0,-10 -x --camera-view-up=1,0,0 --camera-focal-point=1,0,0 --camera-view-angle=20 --camera-azimuth-angle=40 --camera-elevation-angle=-80 --camera-direction=12,34,56 --camera-zoom-factor=78)
 f3d_test(NAME TestCameraDirection DATA suzanne.obj ARGS --camera-direction=-1,-1,1 DEFAULT_LIGHTS)
 f3d_test(NAME TestCameraClipping DATA checkerboard_colorful.obj CONFIG ${F3D_SOURCE_DIR}/testing/configs/checkerboard_colorful.json RESOLUTION 800,600)
+f3d_test(NAME TestOrthographicProjectionCommand ARGS --camera-orthographic DATA cow.vtp DEFAULT_LIGHTS)
```
Comment (mwestphal) [inline application/testing/CMakeLists.txt:157]: TestOrthographicProjectionCommand -> TestCameraOrthographic

File: doc/user/OPTIONS.md [hunk 10]
```diff
@@ -45,7 +45,7 @@ Options|Default|Description
 ------|------|------
 -o, \-\-point-sprites||Show sphere *points sprites* instead of the geometry.
 \-\-point-size=\<size\>|10.0|Set the *size* of points when showing vertices and point sprites.
-\-\-point-type=\<sphere|gaussian\>|sphere|Set the splat type when showing point sprites.
+\-\-point-type=\<sphere>|gaussian\>|sphere|Set the splat type when showing point sprites.
```
Comment (mwestphal) [inline doc/user/OPTIONS.md:48]: wrong change

**Review by mwestphal — APPROVED** [review]

**Review by mwestphal — APPROVED** [review]

**Pull request merged**